Transition Gaea F2 to F5 + Cleanup

FernandoAndrade-NOAA opened the pull request:
This PR addresses #864 and transitions Gaea F2 paths to the newer F5 file system paths as well as reverting to just `gaea`. Detect_machine has been cleaned up in reference to the UFS Weather Model's detect_machine. Adds python load for Hercules to fix call to rt-status.py, resolving errors during the rt.sh final results check.

FernandoAndrade-NOAA committed:
hera rocky8 and hercules python

WenMeng-NOAA committed:
Add changes for Rocky8 on Hera.

WenMeng-NOAA committed:
Add changes for rocky8 on jet.

FernandoAndrade-NOAA committed:
Merge remote-tracking branch 'upstream/post_rocky8' into feature/spack-rocky8

FernandoAndrade-NOAA committed:
gaea f5 & hercules python

revert gaea_c5 to just gaea, update to spack stack 1.6.0, add python load to hercules

FernandoAndrade-NOAA committed:
Resolve gaea lua conflicts

FernandoAndrade-NOAA committed:
update hercules python & cleanup detect_machine

FernandoAndrade-NOAA committed:
Update rt.sh

FernandoAndrade-NOAA committed:
Update detect_machine.sh comment cleanup

WenMeng-NOAA commented on tests/detect_machine.sh:
```diff
@@ -2,99 +2,27 @@
+  gaea5[1-8].ncrc.gov)           MACHINE_ID=gaea ;; ### gaea51-58
 
-  gaea5[0-9]*)             MACHINE_ID=gaea-c5 ;; ### GAEA C5 login node
+  hfe0[1-8])                   MACHINE_ID=hera_c ;; ### hera01-08
```
@FernandoAndrade-NOAA Can you switch hfe05-08 to Rocky8 nodes 'hera' for today's Hera PM?

FernandoAndrade-NOAA replied:
@WenMeng-NOAA updated, thanks for catching that

FernandoAndrade-NOAA committed:
Update detect_machine.sh

aerorahul commented on tests/detect_machine.sh:
Please consider using the `detect_machine.sh` from the ufs-weather-model at https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/detect_machine.sh.
It will allow users to compile the UPP on a compute node of the supported platforms.

FernandoAndrade-NOAA replied:
Thanks for the suggestion @aerorahul, I've gone ahead and adopted the weather model's detect_machine.sh. We will keep UPP's detect_machine.sh in sync with the weather model's moving forward.

aerorahul replied:
Thanks @FernandoAndrade-NOAA

FernandoAndrade-NOAA committed:
Update detect_machine.sh (ufs weather model)

Adopted detect_machine.sh from the ufs-weather-model which should be more flexible in detecting the current machine.

FernandoAndrade-NOAA committed:
rm hera_c & fail message

WenMeng-NOAA commented:
@FernandoAndrade-NOAA Can you start the UPP RTs on Hera and Orion?

WenMeng-NOAA commented:
The UPP RTs were completed on WCOSS2 without changed results.

aerorahul commented on tests/compile_upp.sh:
```diff
@@ -79,28 +79,21 @@ fi
 PATHTR=${PATHTR:-$( cd ${MYDIR}/.. && pwd )}
 source ${PATHTR}/tests/detect_machine.sh
 
+if [[ $MACHINE_ID == "hera_c" ]]; then
+  echo "CentOS will soon be decommissioned on Hera, please utilize Rocky 8 Nodes (hfe05-9 + hfe10-12)."
+  exit 1
+fi
```
The `detect_machine.sh` does not return a `MACHINE_ID = hera_c`.  This block is not necessary.

aerorahul replied:
My bad. I see the `detect_machine.sh` is returning `hera_c` for some `hera` nodes.  That is not the case in the `ufs-weather-model`.

FernandoAndrade-NOAA replied:
Right apologies, I should have mentioned there were some slight modifications to make it clear to users still on hera_c that there is an ongoing transition to the Rocky 8 nodes.

WenMeng-NOAA replied:
@FernandoAndrade-NOAA The hera_c.lua was removed from this PR. No need to keep hera_c detection.

FernandoAndrade-NOAA replied:
@WenMeng-NOAA I've gone ahead and removed hera_c references and a python load to rt.sh which resolved previous errors. Test results before this last commit showed no changes to RT results on Hera, Hercules, and Orion. Gaea showed no issues building.

FernandoAndrade-NOAA committed:
rm hera_c & hercules python load

WenMeng-NOAA commented on ci/rt.sh:
@FernandoAndrade-NOAA Can you update line 68 with the rocky8 spack-stack?

FernandoAndrade-NOAA replied:
Done, would you like me to rerun hera RTs? From my understanding there should still be no changes in results.

WenMeng-NOAA replied:
I just ran RTs on Hera. The results look good for me.

FernandoAndrade-NOAA committed:
Update rt.sh hera rocky8

WenMeng-NOAA commented:
This PR is ready for merging.
